cli/functionary/tokens.py

```diff
@@ -1,8 +1,24 @@
-from .client import post
+import json
 
+import click
+import requests
 
-def login(user: str, password: str):
-    login_response = post(
-        "api-token-auth", data={"username": user, "password": password}
-    )
-    return login_response.get("token")
+
+def login(login_url, user: str, password: str):
+    try:
+        login_response = requests.post(
+            f"{login_url}", data={"username": user, "password": password}
+        )
+        # check status code/message on return then exit
+        if login_response.ok:
+            token = json.loads(login_response.text).get("token")
+            return token
+        else:
```

Comment:
Let's add an `elif login_response.status_code == 400` and have it raise with a more friendly message, like "Unable to login due to an invalid username or password."

This else block is fine for any other (non-400) responses.